pkg/pillar/cmd/zedagent/reportinfo.go

```diff
@@ -669,3 +681,38 @@ func createAppInstances(ctxPtr *zedagentContext,
 	ctxPtr.getconfigCtx.subAppInstanceStatus.Iterate(
 		addAppInstanceFunc)
 }
+
+func getSMARTPowerCycleCountValue() {
+	filename := "/run/SMART_details"
+	data, err := ioutil.ReadFile(filename)
+	if err != nil {
+		log.Errorf("getSMARTPowerCycleCountValue: exception while opening %s. %s", filename, err.Error())
+		powerCycleCountValue = -1
+		return
+	}
+	strData := string(data)
+	if strings.Contains(strData, "SMART support is: Enabled") {
```

Comment:
I am really not a bit fan of unstructured parsing between Go code and shell code.

It makes much more sense for `/run/SMART_details` to have a fixed scheme (simple text in a string or JSON or something similar) and then you can rely on this structure in this piece of code.

This is just a general principle -- we can't expect to audit these types of tools everytime there's an upgrade on the shell side of the tool

Reply:
@rvs if you google smartctl  json you'll find some newer versions support json output. But that isn't available for alpine it seems. If it was I assume we can rely on that more stable json output. If not it is just the smartctl ascii output we have to scrape.

Reply:
smartctl supports `--json` option from 7.0 version (ticket: https://www.smartmontools.org/ticket/766). But smartctl 7.0 version is available only form alpine:3.10 and we are using alpine:3.8 for pillar. 
I was able to dump SMART data in json format after moving the command to storage-init since that uses alpine:3.12.

Reply:
Great!

Reply:
That's why I suggested the move in the first place ;-)